<!DOCTYPE html>
<html lang="en-US">
<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">

<meta name="description" content="Simple minimalist theme">
<meta name="keywords" content="minimalist,blog,goa,hugo,developer">

<base href="/">

<title>Changbum Hong</title>

<meta name="generator" content="Hugo 0.55.5" />


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css">


<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:300,400|Roboto+Slab:400,700|Roboto:300,300i,400,400i,500,500i,700,700i">
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.1.0/css/all.css">
<link rel="stylesheet" href="/css/main.css">
<link rel="stylesheet" href="/css/academicons.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:300,400|Roboto+Slab:400,700|Roboto:300,300i,400,400i,500,500i,700,700i">
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css">
<link rel="stylesheet" href="/css/main.css">
<link rel="stylesheet" href="/css/academicons.css">




<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" href="/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/favicon-16x16.png" sizes="16x16">
<link rel="manifest" href="/manifest.json">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
<meta name="theme-color" content="#ffffff">

</head>
<body lang="en-US">
<div class="container">

<header class="row text-center header">
   <img src="/img/hongiiv2.png" alt="Author Image" class="img-circle text-center headshot"> 
  <h1 class="author">Changbum Hong</h1>
</header>

<section id="info-pane" class="row text-center info">
  
  <h3 class="intro">Software engineer for cancer research  Seoul, Korea.</h3>
  
  
  <h4 class="description">Cancer data science, clinical bioinformatics, DNA variation, precision oncology</h4>
  
</section>

<section id="social-pane" class="row text-center social">
  
  <a href="https://twitter.com/hongiiv" aria-label="Twitter" target="_blank"><i class="fab fa-twitter" aria-hidden="true"></i></a>
  
  
  
  
  
  <a href="https://facebook.com/hongiiv" aria-label="Facebook" target="_blank"><i class="fab fa-facebook" aria-hidden="true"></i></a>
  
  
  <a href="https://github.com/hongiiv" aria-label="Github" target="_blank"><i class="fab fa-github" aria-hidden="true"></i></a>
  
  
  <a href="https://instagram.com/hongiiv" aria-label="Instagram" target="_blank"><i class="fab fa-instagram" aria-hidden="true"></i></a>
  
  
  <a href="https://linkedin.com/in/hongiiv" aria-label="LinkedIn" target="_blank"><i class="fab fa-linkedin" aria-hidden="true"></i></a>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <a href="mailto:hongiiv@gmail.com" aria-label="Email"><i class="fas fa-envelope" aria-hidden="true"></i></a>
  
  
</section>

<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Person",
  "name": "Changbum Hong",
  "url": "",
  "sameAs": [
    "https://twitter.com/hongiiv",
    
    
    
    "https://facebook.com/hongiiv",
    "https://github.com/hongiiv",
    "https://instagram.com/hongiiv",
    "https://linkedin.com/in/hongiiv",
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    ,
    ""
  ]
}
</script>



<header class="row text-left title">
  <div align='center'><h3 class="title">Variants Annotataion</h3></div>
</header>
<section id="category-pane" class="row meta">
  
  <div class="col-md-12">
    <h6 class="text-left meta tcenter">
        PUBLISHED ON MAY 7, 2019 
      
    </h6>
  </div>
  
</section>
<section id="content-pane" class="row">
  <div class="col-md-12 text-justify content">
    

<h2 id="annotation">Annotation</h2>

<p>Variant annotation은 genome sequencing 데이터를 분석하는데에 있어서 가장 마지막 부분에 해당하는 중요한 단계이다. 이 부분에서 부정확하거나 모호한 결과는 곧 질병 연구나 진단에 있어서 문제를 발생시기게 된다. 뭐 예전이야 유전자 단위로 연구하던 시대에는 manual curation으로도 되는 상황이었지만 NGS 시대가 도래하면서 이러한 짓은 더이상 유용하지 않게 되었다.</p>

<p>Variant annotation에서 문제가 발생하는 요인은 여러가지 있을 수 있지만, 다음의 3가지 정도로 요약 정리해 볼 수 있다.</p>

<ol>
<li>Normalization: multi-allelic, Indel left/right alignment</li>
<li>Transcript set: RefSeq, UCSC, Ensemble</li>
<li>Software tool: ANNOVAR, VEP, SNPEFF<br /></li>
</ol>

<h2 id="multi-allelic">Multi-Allelic</h2>

<p>A/C heterozygous 변이(A가 reference)가 있는 경우 우리는 A/C het 또는 C/C homo를 볼 수 있다. 그런데 population 상에서 A/T나 C/T(둘다 non-ref)가 나오고 이러한 경우 variant caller는 VCF 파일에 A/C/T라고 표시하게 된다.</p>

<p>좀 더 어려운 예를 보면 ExAC에서 제공하는 VCF 상에서는 ATC/TTC/A라고 표시된 경우 TTC,A 두개의 alternates가 존재하는데 multi-allelic split과 normalize를 수행하게 된다면 두개의 보정된 변이 A/T와 TC/-(deletion of a TC)가 된다.</p>

<figure>
<img src="/images/2016-09-13/p3.png" style="margin-right:auto; margin-left:auto; width:650px;" />
</figure> 

<h2 id="allelic-primitives-and-left-align">Allelic Primitives and Left Align</h2>

<p>Allelic Primitives: Split multi-nucleotide polymorphisms(MNPs) into individual bases and insertions and deletions.</p>

<p>Left Align: Shift variants to their left-most representation on the reference genome(using a Smith-Waterman local realignment)</p>

<h2 id="software-tool">Software tool</h2>

<p>우리가 흔히 annotation을 수행한다고 한다면 생각나는 tool들은 <a href="http://annovar.openbioinformatics.org/en/latest/">Annovar</a>, <a href="http://snpeff.sourceforge.net/">SnpEff</a>, <a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor (VEP)</a> 정도일 것이다. 그런데 이들이 모두 동일한 결과를 보장하지 않는다. 좀 지났지만 2014년 <a href="https://sites.google.com/site/davismcc/">Davis McCarthy</a> et al. 의 <a href="http://genomemedicine.biomedcentral.com/articles/10.1186/gm543">논문</a>에 따르면 어떠한 툴을 어떠한 transcript 세트를 이용했는지에 따라서 variant의 classification에 영향을 준다고 한다.</p>

<p>이외에도 많은 요소들이 있을 수 있으나 간단히 정리하자면 다음과 같다.</p>

<ol>
<li>HGVS is a standard that is not computable</li>
<li>Public data curation</li>
<li>Versioning</li>
</ol>

<h2 id="artificial-set-of-variants">Artificial set of variants</h2>

<p>BRCA1 유전자의 모든 위치에 해당하는 variant를 인위적으로 생성하여 사용한다.</p>

<ol>
<li>모든 가능한 SNPs</li>
<li>모든 위치의 1 base insertion</li>
<li>모든 위치의 2 base insertion</li>
<li>모든 위치의 3 base insertion</li>
<li>모든 위치의 1,2,3 base deletions</li>
</ol>

<figure>
<img src="/images/2016-09-13/p4.png" style="margin-right:auto; margin-left:auto; width:650px;" />
</figure> 

<h2 id="acmg-guidelines">ACMG Guidelines</h2>

<p>Five-tier terminolgy sytem:</p>

<ol>
<li>&ldquo;pathogenic&rdquo;, &ldquo;likely pathogenic&rdquo;, &ldquo;uncertain significance&rdquo;, &ldquo;likely benign&rdquo;, and &ldquo;benign&rdquo;</li>
<li>Mendelian and mitochondrial variants</li>
<li>Variant assessment guidelines as combined from 11 labs</li>
<li>Report variant with condition and inheritance pattern (c.1521_1423delCTT (p.Phe508del))</li>
<li>Likely pathogenic, likely benign mean 90% certainty</li>
<li>Provide genomic coordinates(g.)</li>
<li>Transcript selection up to lab to define &ldquo;clinically relevant&rdquo;</li>
</ol>

<h2 id="refseq-accession-not-sufficient-for-var-tx-interaction">RefSeq Accession Not Sufficient for Var-Tx Interaction</h2>

<p>RefSeq defines transcripts as mRNA sequence</p>

<ol>
<li>NCBI &ldquo;Annotation Releases&rdquo; (like v105) provides alignments using &ldquo;Splign&rdquo;</li>
<li>UCSC pulls RefSeq mRNA and aligns themselves using &ldquo;BLAT&rdquo;</li>
<li>They can choose equally valid but differnt alignments for the sam accession</li>
<li>This alignment of NM_052814.3 places the exon at dramatically different loci.</li>
<li>Will result in different annotations of any variant overlapping these exons</li>
</ol>

<figure>
<img src="/images/2016-09-13/p1.png" style="margin-right:auto; margin-left:auto; width:650px;" />
</figure> 

<figure>
<img src="/images/2016-09-13/p2.png" style="margin-right:auto; margin-left:auto; width:650px;" />
</figure> 

<p>(crucial step)요즘 회사들은 그들의 제품을 만드는 과정에서 필요한 주요 소프트웨어들을 공개하거나 기술블로그를 운영하고  있다. 그 사례는 비단 구글이나 페이스북과 같이 큰 회사들뿐만 아니라 Biology 분야도 마찬가지인데 그중 몇 개를 우선 소개하고자 한다.</p>

<p>Counsyl은 <a href="https://www.counsyl.com/dev/">https://www.counsyl.com/dev/</a> 를 통해 그들의 소프트웨어를 공개하고 있는데 그중 하나가 HGVS: HGVS variant name parsing and generation <a href="https://github.com/counsyl/hgvs">https://github.com/counsyl/hgvs</a> 이라는 소프트웨어이다.</p>

<pre><code class="language-sh">chr17   41243351    .   A   G   .   .   EFF=intron_variant(MODIFIER|||c.4096+101T&gt;C|1884|BRCA1|protein_coding|CODING|NM_007300.3|10|G),intron_variant(M
ODIFIER|||c.788-302T&gt;C|759|BRCA1|protein_coding|CODING|NM_007298.3|9|G),intron_variant(MODIFIER|||n.4232+101T&gt;C||BRCA1||CODING|NR_027676.1|10|G),intron_variant(MODIFIER|||c.39
55+101T&gt;C|1816|BRCA1|protein_coding|CODING|NM_007297.3|9|G),intron_variant(MODIFIER|||c.788-302T&gt;C|699|BRCA1|protein_coding|CODING|NM_007299.3|10|G),intron_variant(MODIFIER|||
c.4096+101T&gt;C|1863|BRCA1|protein_coding|CODING|NM_007294.3|10|G)    GT  0/1
</code></pre>

<pre><code>&gt;&gt;&gt; snp_eff_ensembl_df=pd.read_table(&quot;BRCA1_exon10_artifical_variant.snpeff.eff.vcf&quot;,skiprows=9,header=0,usecols=range(10))
&gt;&gt;&gt; snp_eff_ensembl_df.head()
  #CHROM       POS ID REF ALT QUAL FILTER  \
0  chr17  41243351  .   A   G    .      .
1  chr17  41243351  .   A   C    .      .
2  chr17  41243351  .   A   T    .      .
3  chr17  41243351  .   A  AA    .      .
4  chr17  41243351  .   A  AG    .      .

                                                INFO FORMAT SAMPLE
0  EFF=intron_variant(MODIFIER|||c.4096+101T&gt;C|18...     GT    0/1
1  EFF=intron_variant(MODIFIER|||c.4096+101T&gt;G|18...     GT    0/1
2  EFF=intron_variant(MODIFIER|||c.4096+101T&gt;A|18...     GT    0/1
3  EFF=intron_variant(MODIFIER|||c.4096+100_4096+...     GT    0/1
4  EFF=intron_variant(MODIFIER|||c.4096+100_4096+...     GT    0/1
</code></pre>

<p>apply(pd.Series,1)</p>

<pre><code>                                                   0  \
0  intron_variant(MODIFIER|||c.4096+101T&gt;C|1884|B...
1  intron_variant(MODIFIER|||c.4096+101T&gt;G|1884|B...
2  intron_variant(MODIFIER|||c.4096+101T&gt;A|1884|B...
3  intron_variant(MODIFIER|||c.4096+100_4096+101i...
4  intron_variant(MODIFIER|||c.4096+100_4096+101i...

                                                   1  \
0  intron_variant(MODIFIER|||c.788-302T&gt;C|759|BRC...
1  intron_variant(MODIFIER|||c.788-302T&gt;G|759|BRC...
2  intron_variant(MODIFIER|||c.788-302T&gt;A|759|BRC...
3  intron_variant(MODIFIER|||c.788-303_788-302ins...
4  intron_variant(MODIFIER|||c.788-303_788-302ins...
</code></pre>

<p>apply(pd.Series,1).stack()</p>

<pre><code>0  0    intron_variant(MODIFIER|||c.4096+101T&gt;C|1884|B...
   1    intron_variant(MODIFIER|||c.788-302T&gt;C|759|BRC...
   2    intron_variant(MODIFIER|||n.4232+101T&gt;C||BRCA1...
   3    intron_variant(MODIFIER|||c.3955+101T&gt;C|1816|B...
   4    intron_variant(MODIFIER|||c.788-302T&gt;C|699|BRC...
   5    intron_variant(MODIFIER|||c.4096+101T&gt;C|1863|B...
1  0    intron_variant(MODIFIER|||c.4096+101T&gt;G|1884|B...
   1    intron_variant(MODIFIER|||c.788-302T&gt;G|759|BRC...
   2    intron_variant(MODIFIER|||n.4232+101T&gt;G||BRCA1...
   3    intron_variant(MODIFIER|||c.3955+101T&gt;G|1816|B...
</code></pre>

  </div>
</section>
<section id="tag-pane" class="row meta">
  
  <div class="col-md-12">
    <h6 class="text-right meta">
      
    </h6>
  </div>
  
</section>








<section id="menu-pane" class="row menu text-center">
  
  
  <span><a class="menu-item" href="/blog/2016-04-26-visualizing-clinvar/">&lt; prev | </a></span>
  
  
  <span><a class="menu-item" href="/blog">blog</a></span>
  
  
  <span><a class="menu-item" href="/blog/2016-04-25-variant-annotation/"> | next &gt;</a></span>
  
  
  <h4 class="text-center"><a class="menu-item" href="/">home</a></h4>
</section>



<footer class="row text-center footer">
  <hr />
  
  <h6 class="text-center copyright">© 2016. Changbum Hong.</h6>
  
  <h6 class="text-center powered">Powered by <a href="https://gohugo.io/">Hugo</a> &amp; <a href="https://github.com/shenoybr/hugo-goa">Goa</a>.</h6>
  
  
</footer>

</div>



<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  

<script type="text/javascript">
hljs.initHighlightingOnLoad();
</script>




<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
<script src="js/main.js"></script>
</body>
</html>


